### PR TITLE
Set GlobalOpenTelemetry to noop

### PR DESCRIPTION
**Pull request by Yas-Franco**
For issue #23923 

- GlobalOpenTelmetry is always returned as OpenTelemetry.noop()
- Users are warned when getting the GlobalOpenTelemetry object
- Users are unable to set the GlobalOpenTelemetry object themselves

### CODE REVIEW

**Comment by Yas-Franco** [comment]
#libby
#build

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java]: 2023

**Review by Azquelt — COMMENTED** [review]
The code here looks good :+1:

We have an odd merge commit, merging integration into this branch, is that necessary?

We also want to output a warning if the user tries to use the `GlobalOpenTelemetry` object.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java
Comment (Yas-Franco) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java]: Done!

**Comment by Yas-Franco** [comment]
I have updated the copyright, removed the unnecessary commit and fixed the bundle imports that was causing test failures. A warning message still needs adding

#libby
#build

**Comment by Yas-Franco** [comment]
#build
#libby

File: dev/io.openliberty.io.opentelemetry.internal/src/io/opentelemetry/api/GlobalOpenTelemetry.java
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/src/io/opentelemetry/api/GlobalOpenTelemetry.java:46]: I remembered that we use `GlobalOpenTelemetry.get()` to get the `OpenTelemetry` instance when using the agent, so we may see the Agent tests fail.

They may not fail, I'm not sure how the classloaders work when the agent is attached and which version of the class will be visible to our code. This version may be hidden completely by the agent in which case it would work.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryGlobalOpenTelemetryTest.java]: `waitForStringInLogUsingMark` returns `null` if the string is not found. This line will throw a `NullPointerException` if the warning isn't emitted.

**Review by Azquelt — APPROVED** [review]
I've left a couple of comments but I think this can be merged as-is as long as the tests pass and any tidying up done later.

Obviously if the tests fail, we need to rethink the approach.

**Comment by Azquelt** [comment]
This also needs L2 and ID review before it can be merged.

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]: Ok, I didn't reveiw this properly, I think we need a bit more explanation here:

I think we can include in the message what has happened and what the user should do instead.

```suggestion
CWMOT5000.cannot.get.globalopentelemetry=CWMOT5000W: The GlobalOpenTelemetry.get method was called. This method returns a non-functional OpenTelemetry object. Use CDI to inject an OpenTelemetry object instead.
CWMOT5000.cannot.get.globalopentelemetry.explanation=There is no global OpenTelemetry object and so using GlobalOpenTelemetry is not supported. Each application must instead use CDI to inject the OpenTelemetry object for that application.
CWMOT5000.cannot.get.globalopentelemetry.useraction=Use CDI to inject the OpenTelemetry object.
```

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]:
```suggestion
CWMOT5001.cannot.set.globalopentelemetry=CWMOT5001E: Setting GlobalOpenTelemetry is not supported
CWMOT5001.cannot.set.globalopentelemetry.explanation=There is no global OpenTelemetry object and so calling the GlobalOpenTelemetry.set method is not supported. Each application must instead use CDI to inject the OpenTelemetry object for that application.
CWMOT5001.cannot.set.globalopentelemetry.useraction=Do not call the GlobalOpenTelemetry.get method. Where the application code needs to obtain an OpenTelemetry object, use CDI to inject it.
```

**Review by Azquelt — CHANGES_REQUESTED** [review]

**Comment by Yas-Franco** [comment]
#build
#libby

**Comment by posmith** [comment]
L2 message review completed.

**Comment by Yas-Franco** [comment]
#build
#libby

**Review by Azquelt — DISMISSED** [review]

**Comment by Azquelt** [comment]
This is ready to merge as soon as it's been reviewed by ID @OpenLiberty/message-reviewer

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (dmuelle) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]:
```suggestion
CWMOT5000.cannot.get.globalopentelemetry.explanation= The GlobalOpenTelemetry class is not supported because no global OpenTelemetry object exists. Instead, each application must use CDI to inject the OpenTelemetry object for that application.
```

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (dmuelle) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]:
```suggestion
CWMOT5001.cannot.set.globalopentelemetry.explanation= The GlobalOpenTelemetry class is not supported because no global OpenTelemetry object exists. Instead, each application must use CDI to inject the OpenTelemetry object for that application.
```

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (dmuelle) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]:
```suggestion
CWMOT5001.cannot.set.globalopentelemetry=CWMOT5001E: Setting the GlobalOpenTelemetry class is not supported.
```

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (dmuelle) [reply dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]: Avoid using "There is..." construction as it obscures subject of sentence and can cause trouble for translators

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- let me know if you have any questions or if I've misinterpreted anything. Thanks

**Comment by Yas-Franco** [comment]
Thanks @dmuelle  - The changes are up 
 
 Rebased with [yasmin-aumeeruddy:telemetry-warning-tracing](https://github.com/OpenLiberty/open-liberty/pull/24397) to prevent merge conflicts 
 
#libby
#build

**Review by dmuelle — DISMISSED** [review]
LGTM thanks

File: dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry.internal/resources/io/openliberty/io/opentelemetry/resources/MPTelemetry.nlsprops]: I agree with the restructuring suggested by ID, but we should have "object" rather than "class" here.
```suggestion
CWMOT5001.cannot.set.globalopentelemetry=CWMOT5001E: Setting the GlobalOpenTelemetry object is not supported.
```

**Comment by Yas-Franco** [comment]
#build
#libby

**Review by Azquelt — APPROVED** [review]

**Pull request merged**
